quickannotator/api/v1/tile/helper.py

```diff
@@ -76,37 +38,53 @@ def tile_intersects_mask_shapely(image_id: int, annotatation_class_id: int, tile
+    tileids, _, _ = get_tile_ids_intersecting_mask(image_id, annotation_class_id, mask_dilation=MASK_DILATION)
     return tile_id in set(tileids)
 
 def get_tile_ids_intersecting_mask(image_id: int, annotation_class_id: int, mask_dilation: int) -> tuple[list, np.ndarray, list]:
     # This function operates in the base magnification space
-    image = get_image_by_id(image_id)
     mask_work_to_base_scale_factor = 1 / base_to_work_scaling_factor(image_id=image_id, annotation_class_id=MASK_CLASS_ID)
-    base_tilesize = get_annotation_class_by_id(annotation_class_id).work_tilesize / base_to_work_scaling_factor(image_id=image_id, annotation_class_id=annotation_class_id)
     
-    # Load GeoJSON mask (assuming polygon)
+    # Get the mask geojson polygons
     model = create_dynamic_model(build_annotation_table_name(image_id, MASK_CLASS_ID, is_gt=True))
-    mask_geojson = get_annotation_query(model, mask_work_to_base_scale_factor).all()    # At work_mag by default
+    mask_geojson: geojson.Polygon = [geojson.loads(ann.polygon) for ann in get_annotation_query(model, mask_work_to_base_scale_factor).all()]    # Scales mask to base mag
 
-    polygons = []
+    tile_ids, mask, processed_polygons = get_tile_ids_intersecting_polygons(image_id, annotation_class_id, mask_geojson, mask_dilation)
 
-    scale_to_tilespace = 1/base_tilesize
-    for annotation in mask_geojson:
-        shapely_polygon = shapely.from_geojson(annotation.polygon)
-        scaled_polygon = scale(shapely_polygon, xfact=scale_to_tilespace, yfact=scale_to_tilespace, origin=(0, 0))
-        polygons.append(np.floor(scaled_polygon.exterior.coords).astype(np.int32))
+    return tile_ids, mask, processed_polygons
 
+def get_tile_ids_intersecting_polygons(image_id: int, annotation_class_id: int, base_polygons: list[geojson.Polygon], mask_dilation: int):
+    """
+    Get tile IDs that intersect with given polygons for a specific image and annotation class.
+    Args:
+        image_id (int): The ID of the image.
+        annotation_class_id (int): The ID of the annotation class.
+        base_polygons (list[geojson.Polygon]): A list of polygons in GeoJSON format. These polygons should be in the base magnification space.
+        mask_dilation (int): The number of iterations for mask dilation.
+    Returns:
+        tuple: A tuple containing:
+            - tile_ids (list[int]): A list of tile IDs that intersect with the polygons.
+            - mask (np.ndarray): The binary mask image with filled polygons.
+            - processed_polygons (list[np.ndarray]): A list of processed polygons with scaled coordinates.
+    """
+
+    image = get_image_by_id(image_id)
+    base_tilesize = get_annotation_class_by_id(annotation_class_id).work_tilesize / base_to_work_scaling_factor(image_id=image_id, annotation_class_id=annotation_class_id)
+    processed_polygons = []
+    scale_factor = 1 / base_tilesize
```

Comment:
do you want to put the scaling in the DB instead of in the backend?

Reply:
These polygons are coming from the client and not the database, so the scaling has to happen in the backend. Do you see another way?